tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/AlternateLocaleData.java

```diff
@@ -0,0 +1,126 @@
+ * based on the {@code "alt"} path attribute (e.g. selecting the short form of a region name).
+ */
+public final class AlternateLocaleData {
+    /**
+     * Returns a wrapped data supplier which will transform any {@link CldrValue}s according to the
+     * supplied {@link CldrPath} mapping. Keys in the path map are the "target" paths of values to
+     * be modified, and the values in the map are the "source" paths from which the replacement
+     * values are obtained. For each map entry, the target and source paths must be in the same
+     * namespace (i.e. have the same path element names).
+     */
+    public static CldrDataSupplier transform(CldrDataSupplier src, Map<CldrPath, CldrPath> altPaths) {
+        return new CldrDataFilter(src, altPaths);
+    }
+
+    private static final class CldrDataFilter extends CldrDataSupplier {
+        private final CldrDataSupplier src;
+        // Mapping from target (destination) to source path. This is necessary since two targets
+        // could come from the same source).
+        private final ImmutableMap<CldrPath, CldrPath> altPaths;
+
+        CldrDataFilter(
+            CldrDataSupplier src, Map<CldrPath, CldrPath> altPaths) {
+            this.src = checkNotNull(src);
+            this.altPaths = ImmutableMap.copyOf(altPaths);
+            altPaths.forEach((t, s) -> checkArgument(hasSameNamespace(checkLdml(t), checkLdml(s)),
+                "alternate paths must have the same namespace: target=%s, source=%s", t, s));
+        }
+
+        @Override
+        public CldrDataSupplier withDraftStatusAtLeast(CldrDraftStatus draftStatus) {
+            return new CldrDataFilter(src.withDraftStatusAtLeast(draftStatus), altPaths);
+        }
+
+        @Override
+        public CldrData getDataForLocale(String localeId, CldrResolution resolution) {
+            return new AltData(src.getDataForLocale(localeId, resolution));
+        }
+
+        @Override
+        public Set<String> getAvailableLocaleIds() {
+            return src.getAvailableLocaleIds();
+        }
+
+        @Override
+        public CldrData getDataForType(CldrDataType type) {
+            return src.getDataForType(type);
+        }
+
+        private final class AltData extends FilteredData {
+            AltData(CldrData srcData) {
+                super(srcData);
+            }
+
+            @Override
+            protected CldrValue filter(CldrValue value) {
+                CldrPath altPath = altPaths.get(value.getPath());
+                if (altPath != null) {
+                    CldrValue altValue = getSourceData().get(altPath);
+                    if (altValue != null) {
+                        return altValue.replacePath(value.getPath());
+                    }
+                }
+                return value;
+            }
+        }
+    }
+
+    private static boolean hasSameNamespace(CldrPath x, CldrPath y) {
```

Comment:
Enforces the requirement that you cannot swap different paths (which would leave CldrValues potentially broken since value attributes would then not be associated with an element on their path).